pkg/dmesg/watcher.go

```diff
@@ -221,3 +241,34 @@ func findISOTimestampIndex(s string) int {
 	}
 	return loc[0]
 }
+
+func (l LogLine) cacheKey() string {
+	h := sha256.New()
+
+	// dedup by second
+	h.Write([]byte(fmt.Sprintf("%d", l.Timestamp.Unix())))
+
+	h.Write([]byte(l.Content))
+	return fmt.Sprintf("%x", h.Sum(nil))
+}
+
+type deduper struct {
+	cache *cache.Cache
+}
+
+func newDeduper(cacheExpiration time.Duration, cachePurgeInterval time.Duration) *deduper {
+	return &deduper{
+		cache: cache.New(cacheExpiration, cachePurgeInterval),
+	}
+}
+
+// isCached returns true if the log line is already in the cache.
+func (d *deduper) isCached(l LogLine) bool {
+	_, found := d.cache.Get(l.cacheKey())
+	return found
+}
+
+// addCache adds the log line to the cache.
+func (d *deduper) addCache(l LogLine) {
+	d.cache.Set(l.cacheKey(), struct{}{}, cache.DefaultExpiration)
```

Comment:
we can record the times to the value, and print it when evicting.

Reply:
done